### PR TITLE
[14.0][MIG][IMP] sale_order_type

### PR DESCRIPTION
**Pull request by BT-cserra**
This PR builds on https://github.com/OCA/sale-workflow/pull/1425

This PR attends the comments made on https://github.com/OCA/sale-workflow/pull/1425, including adding the commits of the two following PRs: https://github.com/OCA/sale-workflow/pull/1531 and https://github.com/OCA/sale-workflow/pull/1553. It also deletes the README so that it is re-generated.

### CODE REVIEW

**Review by oihane — APPROVED** [review]
LGTM

File: sale_order_type/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            line_vals = {}
+            line_vals.update({"route_id": order_type.route_id.id})
+            order.order_line.update(line_vals)
+
+    @api.model
+    def create(self, vals):
+        if vals.get("name", "/") == "/" and vals.get("type_id"):
+            sale_type = self.env["sale.order.type"].browse(vals["type_id"])
+            if sale_type.sequence_id:
+                vals["name"] = sale_type.sequence_id.next_by_id()
```
Comment (ioans73) [inline sale_order_type/models/sale.py]: I think this module is not working properly with the subsequences by date range.
I configure the subsequences for sale orders and, when I create once for the next year, it assings the 2021 sequence.

sale_order_type_range_sequence

Comment (ioans73) [reply sale_order_type/models/sale.py]: ping @BT-cserra

Comment (BT-cserra) [reply sale_order_type/models/sale.py]: @ioans73 did you test with V13 or with this PR?

Comment (ioans73) [reply sale_order_type/models/sale.py]: @BT-cserra with this PR

Comment (BT-cserra) [reply sale_order_type/models/sale.py]: Many thanks @ioans73 you were 100% right! There was a missing parameter. [This commit](https://github.com/OCA/sale-workflow/pull/1646/commits/d53ccc6af7a5d70910c01c28bd49d843dc58d523) fixes it. I've requested a re-review.

Comment (BT-cserra) [reply sale_order_type/models/sale.py]: Wait @ioans73 I forgot to pass the tests. Ouch.

Comment (BT-cserra) [reply sale_order_type/models/sale.py]: I force pushed with https://github.com/OCA/sale-workflow/pull/1646/commits/d53ccc6af7a5d70910c01c28bd49d843dc58d523. Now the checks are in green.

Comment (ioans73) [reply sale_order_type/models/sale.py]: Perfect, thank you very much @BT-cserra

**Review by ioans73 — APPROVED** [review]

**Review by newtratip — APPROVED** [review]

**Comment by newtratip** [comment]
@BT-cserra, ready to merge ?

**Comment by BT-cserra** [comment]
> @BT-cserra, ready to merge ?

@newtratip yes.

**Pull request merged**